data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1800,21 +1800,27 @@ public QueryResult buildOrderBy(String query, @NonNull PersistentEntity entity,
             if (ignoreCase) {
                 buff.append("LOWER(");
             }
-            if (path.getAssociations().isEmpty()) {
+            List<Association> associations = new ArrayList<>(path.getAssociations());
+            int assocCount = associations.size();
+            // If last association is embedded, it does not need to be joined to the alias since it will be in the destination table
+            if (assocCount > 0 && associations.get(assocCount - 1) instanceof Embedded) {
```

Comment:
Should we just filter out all embedded?

Reply:
I did like that initially, but not sure if there are cases where embedded could be somewhere in the middle in the chain of associations, probably not possible. But thought this was safer fix for this case.